ehsm_kms_service/test/test_kms_with_rest.py

```diff
@@ -15,17 +14,20 @@
 
 def test_params(payload):
     params = OrderedDict()
-
     params["appid"] = appid
-    params["nonce"] = str(int.from_bytes(os.urandom(16), "big"))
-    params["timestamp"] = int(time.time())
-
-    query_string = urllib.parse.urlencode(params)
-    query_string += '&app_key=' + appkey
-    sign = str(base64.b64encode(hmac.new(appkey.encode('utf-8'), query_string.encode('utf-8'), digestmod=sha256).digest()),'utf-8')
-
-    params["sign"] = sign
+    params["nonce"] = "".join(random.choice("0123456789") for i in range(16))
+    params["payload"] = urllib.parse.unquote(urllib.parse.urlencode(payload))
+    msec = str(int(time.time()*1000*1000%1000))
+    if(len(msec) == 1):
+        msec = '00' + msec
+    elif(len(msec) == 2):
+        msec = '0' + msec
+    params["timestamp"] = str(time.strftime("%Y%m%d%H%M%S", time.localtime())) + msec
```

Comment:
is timestamp really required? In js layer, the timestamp is generated by the server.

Reply:
It will be fix.